### PR TITLE
azure: add PPS support for azure-proxy-agent

### PR DESCRIPTION
**Pull request by KsenijaS**
## Proposed Commit Message

```
feat(azure): add PPS support for azure-proxy-agent

This PR adds PPS support for azure-proxy agent and improves error logging.

```

## Merge type

- [x] Squash merge using "Proposed Commit Message"
- [ ] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

**Review by cjp256 — APPROVED** [review]
LGTM, thanks!  I think @blackboxsw will like the new logs :D Unit tests need updating though.

File: cloudinit/sources/DataSourceAzure.py [hunk 1]
```diff
@@ -706,6 +722,9 @@ def crawl_metadata(self):
                 self._wait_for_pps_unknown_reuse()
 
             md, userdata_raw, cfg, files = self._reprovision()
+            if cfg.get("ProvisionGuestProxyAgent"):
+                self._check_azure_proxy_agent_status()
```
Comment (blackboxsw) [inline cloudinit/sources/DataSourceAzure.py:726]: No unittests cover this new PPS invocation. If possible can we extend tests/unittests/sources/test_azure::...test_*_pps tests with `construct_ovf_env(provision_guest_proxy_agent=True)` to ensure we get into _check_azure_proxy_agent_status.

**Review by blackboxsw — CHANGES_REQUESTED** [review]
Thanks @KsenijaS for this PR and the improved logs are appreciated. Is there an existing unit test or new unit test we can extend that asserts we check azure proxy agent status in PPS when config is properly set? Then we can ensure we don't degrade this expected behavior in the future.

File: tests/unittests/sources/test_azure.py [hunk 2]
```diff
@@ -4091,13 +4091,144 @@ def test_running_pps(self):
...
+                    provision_guest_proxy_agent=True
+                ).encode()
+            ),
             mock.MagicMock(contents=json.dumps(self.imds_md).encode()),
         ]
         self.mock_azure_get_metadata_from_fabric.return_value = []
 
         self.azure_ds._check_and_get_data()
 
+        assert self.mock_subp_subp.mock_calls == [
```
Comment (cjp256) [inline tests/unittests/sources/test_azure.py:4226]: Awesome, thanks for adding the tests!

For the non-gpa PPS case(s), can we add the counterpoint case (`assert self.mock_subp_subp.mock_calls == []`)?

**Review by cjp256 — APPROVED** [review]
LGTM, thanks for the tests!

**Comment by KsenijaS** [comment]
LGTM! :)

**Review by blackboxsw — APPROVED** [review]
And also LGTM,  Thank you for getting that exhaustive test coverage here.

**Pull request merged**